snapshotter/devmapper/devmapper.go

```diff
@@ -312,18 +310,44 @@ func (dm *Snapshotter) buildMounts(snap storage.Snapshot) []mount.Mount {
 	return mounts
 }
 
-func complete(ctx context.Context, trans storage.Transactor, err error) error {
-	if err == nil {
+// withTransaction wraps fn callback with containerd's meta store transaction.
+// If callback returns an error or transaction is not writable, database transaction will be discarded.
+func (dm *Snapshotter) withTransaction(ctx context.Context, writable bool, fn func(ctx context.Context) error) error {
```

Comment:
would be nice to ensure the behavior of this is tested. As in when an error occurs, the data isn't commited.

Reply:
containerd's snapshotter suite tests cover error scenarios as well (for example [here](https://github.com/containerd/containerd/blob/adfaa697a8b778eb319a49c02406c84bc0f1bdf9/snapshots/testsuite/testsuite.go#L720) and [here](https://github.com/containerd/containerd/blob/adfaa697a8b778eb319a49c02406c84bc0f1bdf9/snapshots/testsuite/testsuite.go#L173)). Not sure it makes a lot of sense to cover same code for our own unit tests.

Reply:
Ah, cool. Wasnt aware.